In [1]:
import pandas as pd
df_class = pd.read_excel(r'C:\Users\Secon\Documents\Disciplinas - Engenharia Mecânica\Projeto de extensão - Isabella\Quadro de horários - automatização\Salas-Controle.xlsx', sheet_name="Obrigatória")
df_class.columns = df_class.iloc[1]
df_class.drop([0,1],axis=0,inplace=True)
df_class.head()

1 Departamento  Dia da semana  Horário Disciplina         Professor  \
2          TEE  segunda-feira  07h/08h   TEE00114         Alexandre   
3          TER  segunda-feira  07h/08h   TER00114  Carlos Rodrigues   
4          TEE  segunda-feira  08h/09h   TEE00114         Alexandre   
5          TER  segunda-feira  08h/09h   TER00040     Rodrigo Amado   
6          TER  segunda-feira  08h/09h   TER00114  Carlos Rodrigues   

1 Vagas ofertadas Média das vagas preenchidas  Sala         Tipo    COMEÇO  \
2              30                          20  233D  Obrigatória  07:00:00   
3              40                          37   NaN  Obrigatória  07:00:00   
4              30                          16  233D  Obrigatória  08:00:00   
5              70                          40   NaN  Obrigatória  08:00:00   
6              40                          33   NaN  Obrigatória  08:00:00   

1       FIM Quantas horas ainda faltam pra acabar  \
2  08:00:00                                     2   
3  08:00:00                                     2   
4  09:00:00                                     1   
5  09:00:00                                     4   
6  09:00:00                                     1   

1 Duração efetiva das aulas(contada a partir da primeira)  
2                                                  2       
3                                                  2       
4                                  Já tinha começado       
5                                                  4       
6                                  Já tinha começado

In [2]:
df_time = pd.read_excel(r'C:\Users\Secon\Documents\Disciplinas - Engenharia Mecânica\Projeto de extensão - Isabella\Horarios_new.xlsx', header=None)

In [3]:
df_time.head()

0        1     2      3     4       5
0  Dia da semana  Horário  Sala  Bloco  Dept  Capac.
1  segunda-feira  07h/08h   104      D   NaN      60
2  segunda-feira  08h/09h   104      D   NaN      60
3  segunda-feira  09h/10h   104      D   NaN      60
4  segunda-feira  10h/11h   104      D   NaN      60

In [4]:
df_class[df_class['Duração efetiva das aulas(contada a partir da primeira)'] != 'Já tinha começado'][['Horário', 'Disciplina', 'Professor',
       'Vagas ofertadas', 'Sala', 'Tipo',
       'COMEÇO', 'FIM', 'Quantas horas ainda faltam pra acabar',
       'Duração efetiva das aulas(contada a partir da primeira)']]['COMEÇO'].dtype

dtype('O')

In [5]:
df_t1 = df_class[df_class['Duração efetiva das aulas(contada a partir da primeira)'] != 'Já tinha começado'][['Departamento','Dia da semana', 'COMEÇO', 'Disciplina', 'Professor',
       'Vagas ofertadas', 'Sala', 'Tipo',
       'Duração efetiva das aulas(contada a partir da primeira)']]

In [6]:
df_t1[(df_t1['Professor'] == 'Alexandre')&(df_t1['Disciplina'] == 'TEE00114')]


1    Departamento  Dia da semana    COMEÇO Disciplina  Professor  \
2             TEE  segunda-feira  07:00:00   TEE00114  Alexandre   
518           TEE   quarta-feira  07:00:00   TEE00114  Alexandre   
1035          TEE    sexta-feira  07:00:00   TEE00114  Alexandre   

1    Vagas ofertadas  Sala         Tipo  \
2                 30  233D  Obrigatória   
518               30  233D  Obrigatória   
1035              30  233D  Obrigatória   

1    Duração efetiva das aulas(contada a partir da primeira)  
2                                                     2       
518                                                   2       
1035                                                  2

In [7]:
df_t1[(df_t1['Professor'] == 'Alexandre')&(df_t1['Disciplina'] == 'TEE00114')]['Dia da semana']


2       segunda-feira
518      quarta-feira
1035      sexta-feira
Name: Dia da semana, dtype: object

Colocar o horário de início também como filtro, de precaução, pois pode ter a mesma disciplina, num dia diferente, num outro dia, com horários diferentes. <b>(Pending)</b>

In [8]:
lista = []
for i in df_t1[(df_t1['Professor'] == 'Alexandre')&(df_t1['Disciplina'] == 'TEE00114')]['Dia da semana']:
    lista.append(i)
print(lista)
lista.clear()
print(lista)

['segunda-feira', 'quarta-feira', 'sexta-feira']
[]


In [9]:
df_t1.columns = ['Departamento', 'Dia da semana', 'COMEÇO', 'Disciplina', 'Professor',
       'Vagas ofertadas', 'Sala', 'Tipo',
       'Duracao']

In [10]:
df_t1.columns

Index(['Departamento', 'Dia da semana', 'COMEÇO', 'Disciplina', 'Professor',
       'Vagas ofertadas', 'Sala', 'Tipo', 'Duracao'],
      dtype='object')

In [11]:
df_t1['Dia da semana'] = df_t1['Dia da semana'].replace('Sábado', 'sábado')

In [12]:
df_t2 = df_t1.copy()

In [13]:
for i in df_t2['Dia da semana'].unique():
    df_t2[i] = ""
df_t2.head()

Departamento  Dia da semana    COMEÇO Disciplina  \
2          TEE  segunda-feira  07:00:00   TEE00114   
3          TER  segunda-feira  07:00:00   TER00114   
5          TER  segunda-feira  08:00:00   TER00040   
7          TEQ  segunda-feira  08:00:00   TEQ00175   
8          TEC  segunda-feira  09:00:00   TEC00224   

                      Professor Vagas ofertadas         Sala         Tipo  \
2                     Alexandre              30         233D  Obrigatória   
3              Carlos Rodrigues              40          NaN  Obrigatória   
5                 Rodrigo Amado              70          NaN  Obrigatória   
7  ROGERIO FERNANDES DE LACERDA              15  LABORATÓRIO  Obrigatória   
8    MARCO AURELIO CHAVES FERRO              45          NaN  Obrigatória   

  Duracao segunda-feira terça-feira quarta-feira quinta-feira sexta-feira  \
2       2                                                                   
3       2                                                                   
5       4                                                                   
7       3                                                                   
8       2                                                                   

  sábado  
2         
3         
5         
7         
8

In [14]:
lista = []
for a in range(df_t2.shape[0]):
    for i in df_t2[(df_t1['Professor'] == df_t2.iloc[a,4])&(df_t1['Disciplina'] == df_t2.iloc[a, 3])&(df_t1['COMEÇO'] == df_t2.iloc[a, 2])]['Dia da semana']:
        lista.append(i)
        index = df_t2[(df_t1['Professor'] == df_t2.iloc[a,4])&(df_t1['Disciplina'] == df_t2.iloc[a, 3])&(df_t1['COMEÇO'] == df_t2.iloc[a, 2])].index
        for c in index:
            for b in lista:
                df_t2.loc[c][b] = 1
        lista.clear()

In [15]:
df_t2

Departamento  Dia da semana    COMEÇO Disciplina  \
2             TEE  segunda-feira  07:00:00   TEE00114   
3             TER  segunda-feira  07:00:00   TER00114   
5             TER  segunda-feira  08:00:00   TER00040   
7             TEQ  segunda-feira  08:00:00   TEQ00175   
8             TEC  segunda-feira  09:00:00   TEC00224   
...           ...            ...       ...        ...   
1131          TER    sexta-feira  18:00:00   TER00059   
1138          TEM         sábado  07:00:00   TEM00214   
1148          TEQ         sábado  16:00:00   TEQ00115   
1150          TET         sábado  18:00:00   TET00309   
1151          TET         sábado  18:00:00   TET00310   

                         Professor Vagas ofertadas         Sala         Tipo  \
2                        Alexandre              30         233D  Obrigatória   
3                 Carlos Rodrigues              40          NaN  Obrigatória   
5                    Rodrigo Amado              70          NaN  Obrigatória   
7     ROGERIO FERNANDES DE LACERDA              15  LABORATÓRIO  Obrigatória   
8       MARCO AURELIO CHAVES FERRO              45          NaN  Obrigatória   
...                            ...             ...          ...          ...   
1131             Ricardo Abranches              60          NaN  Obrigatória   
1138    Leandro Alcoforado Sphaier              45          NaN  Obrigatória   
1148        ALBINO LOPES D ALMEIDA              20         xxxx  Obrigatória   
1150                         Paula              45          NaN  Obrigatória   
1151                       Cledson              45          NaN  Obrigatória   

     Duracao segunda-feira terça-feira quarta-feira quinta-feira sexta-feira  \
2          2             1                        1                        1   
3          2             1                                                     
5          4             1                                                     
7          3             1                                                     
8          2             1                                     1               
...      ...           ...         ...          ...          ...         ...   
1131       4                                                               1   
1138       4                                                                   
1148       2                                                                   
1150       2                                                                   
1151       2                                                                   

     sábado  
2            
3            
5            
7            
8            
...     ...  
1131         
1138      1  
1148      1  
1150      1  
1151      1  

[460 rows x 15 columns]

In [16]:
df_t2[(df_t1['Professor'] == df_t2.iloc[0,4])&(df_t1['Disciplina'] == df_t2.iloc[0, 3])&(df_t1['COMEÇO'] == df_t2.iloc[0, 2])]

Departamento  Dia da semana    COMEÇO Disciplina  Professor  \
2             TEE  segunda-feira  07:00:00   TEE00114  Alexandre   
518           TEE   quarta-feira  07:00:00   TEE00114  Alexandre   
1035          TEE    sexta-feira  07:00:00   TEE00114  Alexandre   

     Vagas ofertadas  Sala         Tipo Duracao segunda-feira terça-feira  \
2                 30  233D  Obrigatória       2             1               
518               30  233D  Obrigatória       2             1               
1035              30  233D  Obrigatória       2             1               

     quarta-feira quinta-feira sexta-feira sábado  
2               1                        1         
518             1                        1         
1035            1                        1

In [17]:
df_t2[(df_t1['Professor'] == df_t2.iloc[0,4])&(df_t1['Disciplina'] == df_t2.iloc[0, 3])&(df_t1['COMEÇO'] == df_t2.iloc[0, 2])]['Dia da semana']

2       segunda-feira
518      quarta-feira
1035      sexta-feira
Name: Dia da semana, dtype: object

In [18]:
df_t2.iloc[0]

Departamento                 TEE
Dia da semana      segunda-feira
COMEÇO                  07:00:00
Disciplina              TEE00114
Professor              Alexandre
Vagas ofertadas               30
Sala                        233D
Tipo                 Obrigatória
Duracao                        2
segunda-feira                  1
terça-feira                     
quarta-feira                   1
quinta-feira                    
sexta-feira                    1
sábado                          
Name: 2, dtype: object

In [19]:
df_t2.duplicated().value_counts()

False    460
dtype: int64

In [20]:
df_t2.drop('Dia da semana', axis=1).duplicated().value_counts()

False    328
True     132
dtype: int64

In [21]:
df_test = df_t2.drop('Dia da semana', axis=1)
df_test.drop_duplicates()

Departamento    COMEÇO Disciplina                     Professor  \
2             TEE  07:00:00   TEE00114                     Alexandre   
3             TER  07:00:00   TER00114              Carlos Rodrigues   
5             TER  08:00:00   TER00040                 Rodrigo Amado   
7             TEQ  08:00:00   TEQ00175  ROGERIO FERNANDES DE LACERDA   
8             TEC  09:00:00   TEC00224    MARCO AURELIO CHAVES FERRO   
...           ...       ...        ...                           ...   
1131          TER  18:00:00   TER00059             Ricardo Abranches   
1138          TEM  07:00:00   TEM00214    Leandro Alcoforado Sphaier   
1148          TEQ  16:00:00   TEQ00115        ALBINO LOPES D ALMEIDA   
1150          TET  18:00:00   TET00309                         Paula   
1151          TET  18:00:00   TET00310                       Cledson   

     Vagas ofertadas         Sala         Tipo Duracao segunda-feira  \
2                 30         233D  Obrigatória       2             1   
3                 40          NaN  Obrigatória       2             1   
5                 70          NaN  Obrigatória       4             1   
7                 15  LABORATÓRIO  Obrigatória       3             1   
8                 45          NaN  Obrigatória       2             1   
...              ...          ...          ...     ...           ...   
1131              60          NaN  Obrigatória       4                 
1138              45          NaN  Obrigatória       4                 
1148              20         xxxx  Obrigatória       2                 
1150              45          NaN  Obrigatória       2                 
1151              45          NaN  Obrigatória       2                 

     terça-feira quarta-feira quinta-feira sexta-feira sábado  
2                           1                        1         
3                                                              
5                                                              
7                                                              
8                                        1                     
...          ...          ...          ...         ...    ...  
1131                                                 1         
1138                                                        1  
1148                                                        1  
1150                                                        1  
1151                                                        1  

[328 rows x 14 columns]

In [22]:
df_t2.shape

(460, 15)

In [23]:
df_final = df_test.drop_duplicates().reset_index(drop=True)

In [24]:
df_final.tail()

Departamento    COMEÇO Disciplina                   Professor  \
323          TER  18:00:00   TER00059           Ricardo Abranches   
324          TEM  07:00:00   TEM00214  Leandro Alcoforado Sphaier   
325          TEQ  16:00:00   TEQ00115      ALBINO LOPES D ALMEIDA   
326          TET  18:00:00   TET00309                       Paula   
327          TET  18:00:00   TET00310                     Cledson   

    Vagas ofertadas  Sala         Tipo Duracao segunda-feira terça-feira  \
323              60   NaN  Obrigatória       4                             
324              45   NaN  Obrigatória       4                             
325              20  xxxx  Obrigatória       2                             
326              45   NaN  Obrigatória       2                             
327              45   NaN  Obrigatória       2                             

    quarta-feira quinta-feira sexta-feira sábado  
323                                     1         
324                                            1  
325                                            1  
326                                            1  
327                                            1

In [25]:
df_time.columns = df_time.iloc[0]
df_time.drop(0, axis=0, inplace=True)
df_time.reset_index(drop=True, inplace=True)

In [26]:
df_time.head()

0  Dia da semana  Horário Sala Bloco Dept Capac.
0  segunda-feira  07h/08h  104     D  NaN     60
1  segunda-feira  08h/09h  104     D  NaN     60
2  segunda-feira  09h/10h  104     D  NaN     60
3  segunda-feira  10h/11h  104     D  NaN     60
4  segunda-feira  11h/12h  104     D  NaN     60

In [27]:
df_time['Sala'].unique()

array([104, 137, '230A', 233, '237B', 238, 239, '307A', 331, 332, '334-D',
       336, 337, 339, 443, '443A', 444, 445, '446-D', 447, 448, 449, 450,
       502, '502C', '531B', '234A', '234B', '334-E', 341, 428, '446-E',
       201, 202, 204, 205], dtype=object)

In [28]:
df_time['Departamento'] = ""
df_time['Disciplina'] = ""
df_time['Professor'] = ""

In [29]:
df_time['Dia da semana'].value_counts()

segunda-feira    540
terça-feira      540
quarta-feira     540
quinta-feira     540
sexta-feira      540
sábado           540
Name: Dia da semana, dtype: int64

In [30]:
df_final.iloc[0][df_final.columns[8]]

1

In [31]:
a = df_final.iloc[150]['COMEÇO']
type(a.hour)

int

In [32]:
def horario(x):
    return int(x[:2])

In [33]:
df_time['Horário']= df_time['Horário'].apply(horario)

In [34]:
df_final.iloc[0]['COMEÇO']

datetime.time(7, 0)

In [35]:
df_time[(df_time['Dia da semana'] == 'segunda-feira')&(df_time['Horário'] == df_final.iloc[0]['COMEÇO'].hour)&(df_time['Sala'] == df_time['Sala'].unique()[0])]

0  Dia da semana  Horário Sala Bloco Dept Capac. Departamento Disciplina  \
0  segunda-feira        7  104     D  NaN     60                           

0 Professor  
0

In [36]:

if df_time[(df_time['Dia da semana'] == 'segunda-feira')&(df_time['Horário'] == df_final.iloc[0]['COMEÇO'].hour)&(df_time['Sala'] == df_time['Sala'].unique()[0])]['Disciplina'].item() == "":
    print('Yes')

Yes


In [37]:
df_final.iloc[1]['COMEÇO']

datetime.time(7, 0)

In [38]:
palavra = df_final.iloc[0]['Disciplina']

In [39]:
df_time.iloc[0, 6]

''

In [40]:
df_time.iloc[0, 6] = df_final.iloc[0]['Disciplina']
print(df_time.iloc[0, 6])

TEE00114


Considerar caso de que, as vezes, uma matéria pode ser ministradas em duas salas diferentes <b>(Solved)</b><br>
Primeiro programar para todas as disciplinas receberem as aulas, respeitando os dois critérios.<br>
Depois disso, verificar se existem algumas salas vagas nos horárias das disciplinas que não consegueram receber salas. Se sim, atribuir as duas salas distintas para essas determinada disciplina que sobrou<br>
Considerar os dois critérios na programação <b>(Pending)</b><br>

In [41]:
df_time.loc[0]['Horário'] == df_final.iloc[1]['COMEÇO'].hour

True

In [71]:
df_tempo = df_time.copy()

In [72]:
df_aulas = df_final.copy()
df_aulas['Local'] = ""

## Bussiness core

Vou modificar o nome das salas 334 para 334-D e 334-E, para distingui-los, já que tem nos dois blocos, com mesmo nome<br>
Mesmo caso para a sala 446: ficará 446-D e 446-E, já que também tem duas com mesmo nome nos dois blocos<br>
Modifiquei isso na planilha Horário_new

In [73]:
salas = df_tempo['Sala'].unique()
disciplinas_fora = []
for i in range(df_aulas.shape[0]):
    # 
    duration = int(df_aulas.iloc[i]['Duracao'])
    begin = df_aulas.iloc[i]['COMEÇO'].hour
    # pegando o primeiro dia da semana que tem aula
    days = []
    for num in range(8, 14):
        if df_aulas.iloc[i][df_aulas.columns[num]] == 1: # Aqui ele está pegando todos os dias da semana que tem aula
            days.append(df_aulas.columns[num])
    print(days)
    # Verificando se a sala está vazia, considerando as condições. Se sim, então atribuir sala à disciplina
    try:
        for n in range(len(salas)):
            if df_tempo[(df_tempo['Dia da semana'] == days[0])&(df_tempo['Horário'] == begin)&(df_tempo['Sala'] == salas[n])]['Disciplina'].item() == "":
                print("begin: ", begin)
                for d in days:
                    begin1 = begin
                    f = 0
                    while f != duration:
                        index = df_tempo[(df_tempo['Dia da semana'] == d)&(df_tempo['Horário'] == begin1)&(df_tempo['Sala'] == salas[n])].index
                        df_tempo.iloc[index, 7] = df_aulas.loc[i]['Disciplina']
                        df_tempo.iloc[index, 6] = df_aulas.loc[i]['Departamento']
                        df_tempo.iloc[index, 8] = df_aulas.loc[i]['Professor']
                        print("begin1: ", begin1)
                        print(index)
                        begin1 = begin1 +1
                        f += 1
                df_aulas.iloc[i, 14] = salas[n] #Acrescentar sala para indicar que determinada disciplina agora tem uma sala
                break
            else:
                print('A sala ', salas[n], "no horário", begin, "do dia ", days[0], 'tá ocupada')
                if n == (len(salas) - 1):
                    disciplinas_fora.append(df_aulas.loc[i]['Disciplina'])
                continue
    except:
        print("Esse filha da puta aqui que atrapalhou", salas[n], begin, days[0])
        raise
    print("Foi")
    days.clear()
print('Disciplinas que ficaram de fora: ', disciplinas_fora)

['segunda-feira', 'quarta-feira', 'sexta-feira']
begin:  7
begin1:  7
Int64Index([0], dtype='int64')
begin1:  8
Int64Index([1], dtype='int64')
begin1:  7
Int64Index([30], dtype='int64')
begin1:  8
Int64Index([31], dtype='int64')
begin1:  7
Int64Index([60], dtype='int64')
begin1:  8
Int64Index([61], dtype='int64')
Foi
['segunda-feira']
A sala  104 no horário 7 do dia  segunda-feira tá ocupada
begin:  7
begin1:  7
Int64Index([75], dtype='int64')
begin1:  8
Int64Index([76], dtype='int64')
Foi
['segunda-feira']
A sala  104 no horário 8 do dia  segunda-feira tá ocupada
A sala  137 no horário 8 do dia  segunda-feira tá ocupada
begin:  8
begin1:  8
Int64Index([151], dtype='int64')
begin1:  9
Int64Index([152], dtype='int64')
begin1:  10
Int64Index([153], dtype='int64')
begin1:  11
Int64Index([154], dtype='int64')
Foi
['segunda-feira']
A sala  104 no horário 8 do dia  segunda-feira tá ocupada
A sala  137 no horário 8 do dia  segunda-feira tá ocupada
A sala  230A no horário 8 do dia  segunda-fei

begin1:  10
Int64Index([978], dtype='int64')
Foi
['segunda-feira', 'quarta-feira']
A sala  104 no horário 9 do dia  segunda-feira tá ocupada
A sala  137 no horário 9 do dia  segunda-feira tá ocupada
A sala  230A no horário 9 do dia  segunda-feira tá ocupada
A sala  233 no horário 9 do dia  segunda-feira tá ocupada
A sala  237B no horário 9 do dia  segunda-feira tá ocupada
A sala  238 no horário 9 do dia  segunda-feira tá ocupada
A sala  239 no horário 9 do dia  segunda-feira tá ocupada
A sala  307A no horário 9 do dia  segunda-feira tá ocupada
A sala  331 no horário 9 do dia  segunda-feira tá ocupada
A sala  332 no horário 9 do dia  segunda-feira tá ocupada
A sala  334-D no horário 9 do dia  segunda-feira tá ocupada
A sala  336 no horário 9 do dia  segunda-feira tá ocupada
A sala  337 no horário 9 do dia  segunda-feira tá ocupada
A sala  339 no horário 9 do dia  segunda-feira tá ocupada
begin:  9
begin1:  9
Int64Index([1052], dtype='int64')
begin1:  10
Int64Index([1053], dtype='int64')

begin1:  15
Int64Index([8], dtype='int64')
begin1:  16
Int64Index([9], dtype='int64')
Foi
['segunda-feira']
A sala  104 no horário 13 do dia  segunda-feira tá ocupada
begin:  13
begin1:  13
Int64Index([81], dtype='int64')
begin1:  14
Int64Index([82], dtype='int64')
begin1:  15
Int64Index([83], dtype='int64')
Foi
['segunda-feira']
A sala  104 no horário 13 do dia  segunda-feira tá ocupada
A sala  137 no horário 13 do dia  segunda-feira tá ocupada
begin:  13
begin1:  13
Int64Index([156], dtype='int64')
begin1:  14
Int64Index([157], dtype='int64')
begin1:  15
Int64Index([158], dtype='int64')
Foi
['segunda-feira']
A sala  104 no horário 14 do dia  segunda-feira tá ocupada
A sala  137 no horário 14 do dia  segunda-feira tá ocupada
A sala  230A no horário 14 do dia  segunda-feira tá ocupada
begin:  14
begin1:  14
Int64Index([232], dtype='int64')
begin1:  15
Int64Index([233], dtype='int64')
begin1:  16
Int64Index([234], dtype='int64')
begin1:  17
Int64Index([235], dtype='int64')
Foi
['segunda

A sala  137 no horário 14 do dia  segunda-feira tá ocupada
A sala  230A no horário 14 do dia  segunda-feira tá ocupada
A sala  233 no horário 14 do dia  segunda-feira tá ocupada
A sala  237B no horário 14 do dia  segunda-feira tá ocupada
A sala  238 no horário 14 do dia  segunda-feira tá ocupada
A sala  239 no horário 14 do dia  segunda-feira tá ocupada
A sala  307A no horário 14 do dia  segunda-feira tá ocupada
A sala  331 no horário 14 do dia  segunda-feira tá ocupada
A sala  332 no horário 14 do dia  segunda-feira tá ocupada
A sala  334-D no horário 14 do dia  segunda-feira tá ocupada
A sala  336 no horário 14 do dia  segunda-feira tá ocupada
A sala  337 no horário 14 do dia  segunda-feira tá ocupada
A sala  339 no horário 14 do dia  segunda-feira tá ocupada
A sala  443 no horário 14 do dia  segunda-feira tá ocupada
begin:  14
begin1:  14
Int64Index([1132], dtype='int64')
begin1:  15
Int64Index([1133], dtype='int64')
begin1:  14
Int64Index([1162], dtype='int64')
begin1:  15
Int64Ind

A sala  332 no horário 14 do dia  segunda-feira tá ocupada
A sala  334-D no horário 14 do dia  segunda-feira tá ocupada
A sala  336 no horário 14 do dia  segunda-feira tá ocupada
A sala  337 no horário 14 do dia  segunda-feira tá ocupada
A sala  339 no horário 14 do dia  segunda-feira tá ocupada
A sala  443 no horário 14 do dia  segunda-feira tá ocupada
A sala  443A no horário 14 do dia  segunda-feira tá ocupada
A sala  444 no horário 14 do dia  segunda-feira tá ocupada
A sala  445 no horário 14 do dia  segunda-feira tá ocupada
A sala  446-D no horário 14 do dia  segunda-feira tá ocupada
A sala  447 no horário 14 do dia  segunda-feira tá ocupada
A sala  448 no horário 14 do dia  segunda-feira tá ocupada
A sala  449 no horário 14 do dia  segunda-feira tá ocupada
begin:  14
begin1:  14
Int64Index([1657], dtype='int64')
begin1:  15
Int64Index([1658], dtype='int64')
begin1:  14
Int64Index([1687], dtype='int64')
begin1:  15
Int64Index([1688], dtype='int64')
Foi
['segunda-feira', 'quarta-fei

begin1:  14
Int64Index([2062], dtype='int64')
begin1:  15
Int64Index([2063], dtype='int64')
Foi
['segunda-feira', 'quarta-feira']
A sala  104 no horário 14 do dia  segunda-feira tá ocupada
A sala  137 no horário 14 do dia  segunda-feira tá ocupada
A sala  230A no horário 14 do dia  segunda-feira tá ocupada
A sala  233 no horário 14 do dia  segunda-feira tá ocupada
A sala  237B no horário 14 do dia  segunda-feira tá ocupada
A sala  238 no horário 14 do dia  segunda-feira tá ocupada
A sala  239 no horário 14 do dia  segunda-feira tá ocupada
A sala  307A no horário 14 do dia  segunda-feira tá ocupada
A sala  331 no horário 14 do dia  segunda-feira tá ocupada
A sala  332 no horário 14 do dia  segunda-feira tá ocupada
A sala  334-D no horário 14 do dia  segunda-feira tá ocupada
A sala  336 no horário 14 do dia  segunda-feira tá ocupada
A sala  337 no horário 14 do dia  segunda-feira tá ocupada
A sala  339 no horário 14 do dia  segunda-feira tá ocupada
A sala  443 no horário 14 do dia  segun

begin1:  17
Int64Index([610], dtype='int64')
begin1:  16
Int64Index([639], dtype='int64')
begin1:  17
Int64Index([640], dtype='int64')
Foi
['segunda-feira', 'quarta-feira']
A sala  104 no horário 16 do dia  segunda-feira tá ocupada
A sala  137 no horário 16 do dia  segunda-feira tá ocupada
A sala  230A no horário 16 do dia  segunda-feira tá ocupada
A sala  233 no horário 16 do dia  segunda-feira tá ocupada
A sala  237B no horário 16 do dia  segunda-feira tá ocupada
A sala  238 no horário 16 do dia  segunda-feira tá ocupada
A sala  239 no horário 16 do dia  segunda-feira tá ocupada
A sala  307A no horário 16 do dia  segunda-feira tá ocupada
A sala  331 no horário 16 do dia  segunda-feira tá ocupada
begin:  16
begin1:  16
Int64Index([684], dtype='int64')
begin1:  17
Int64Index([685], dtype='int64')
begin1:  16
Int64Index([714], dtype='int64')
begin1:  17
Int64Index([715], dtype='int64')
Foi
['segunda-feira', 'quarta-feira']
A sala  104 no horário 16 do dia  segunda-feira tá ocupada
A sal

begin1:  19
Int64Index([87], dtype='int64')
begin1:  18
Int64Index([116], dtype='int64')
begin1:  19
Int64Index([117], dtype='int64')
Foi
['segunda-feira', 'quarta-feira']
A sala  104 no horário 18 do dia  segunda-feira tá ocupada
A sala  137 no horário 18 do dia  segunda-feira tá ocupada
begin:  18
begin1:  18
Int64Index([161], dtype='int64')
begin1:  19
Int64Index([162], dtype='int64')
begin1:  18
Int64Index([191], dtype='int64')
begin1:  19
Int64Index([192], dtype='int64')
Foi
['segunda-feira']
A sala  104 no horário 18 do dia  segunda-feira tá ocupada
A sala  137 no horário 18 do dia  segunda-feira tá ocupada
A sala  230A no horário 18 do dia  segunda-feira tá ocupada
begin:  18
begin1:  18
Int64Index([236], dtype='int64')
begin1:  19
Int64Index([237], dtype='int64')
begin1:  20
Int64Index([238], dtype='int64')
begin1:  21
Int64Index([239], dtype='int64')
Foi
['segunda-feira', 'quinta-feira']
A sala  104 no horário 18 do dia  segunda-feira tá ocupada
A sala  137 no horário 18 do di

begin1:  19
Int64Index([1062], dtype='int64')
begin1:  18
Int64Index([1091], dtype='int64')
begin1:  19
Int64Index([1092], dtype='int64')
Foi
['segunda-feira', 'quarta-feira']
A sala  104 no horário 18 do dia  segunda-feira tá ocupada
A sala  137 no horário 18 do dia  segunda-feira tá ocupada
A sala  230A no horário 18 do dia  segunda-feira tá ocupada
A sala  233 no horário 18 do dia  segunda-feira tá ocupada
A sala  237B no horário 18 do dia  segunda-feira tá ocupada
A sala  238 no horário 18 do dia  segunda-feira tá ocupada
A sala  239 no horário 18 do dia  segunda-feira tá ocupada
A sala  307A no horário 18 do dia  segunda-feira tá ocupada
A sala  331 no horário 18 do dia  segunda-feira tá ocupada
A sala  332 no horário 18 do dia  segunda-feira tá ocupada
A sala  334-D no horário 18 do dia  segunda-feira tá ocupada
A sala  336 no horário 18 do dia  segunda-feira tá ocupada
A sala  337 no horário 18 do dia  segunda-feira tá ocupada
A sala  339 no horário 18 do dia  segunda-feira tá o

begin1:  11
Int64Index([574], dtype='int64')
begin1:  12
Int64Index([575], dtype='int64')
Foi
['terça-feira', 'quinta-feira']
A sala  104 no horário 9 do dia  terça-feira tá ocupada
A sala  137 no horário 9 do dia  terça-feira tá ocupada
A sala  230A no horário 9 do dia  terça-feira tá ocupada
A sala  233 no horário 9 do dia  terça-feira tá ocupada
A sala  237B no horário 9 do dia  terça-feira tá ocupada
A sala  238 no horário 9 do dia  terça-feira tá ocupada
A sala  239 no horário 9 do dia  terça-feira tá ocupada
A sala  307A no horário 9 do dia  terça-feira tá ocupada
begin:  9
begin1:  9
Int64Index([617], dtype='int64')
begin1:  10
Int64Index([618], dtype='int64')
begin1:  9
Int64Index([647], dtype='int64')
begin1:  10
Int64Index([648], dtype='int64')
Foi
['terça-feira', 'quinta-feira']
A sala  104 no horário 9 do dia  terça-feira tá ocupada
A sala  137 no horário 9 do dia  terça-feira tá ocupada
A sala  230A no horário 9 do dia  terça-feira tá ocupada
A sala  233 no horário 9 do di

A sala  443 no horário 9 do dia  terça-feira tá ocupada
A sala  443A no horário 9 do dia  terça-feira tá ocupada
A sala  444 no horário 9 do dia  terça-feira tá ocupada
begin:  9
begin1:  9
Int64Index([1292], dtype='int64')
begin1:  10
Int64Index([1293], dtype='int64')
begin1:  11
Int64Index([1294], dtype='int64')
begin1:  12
Int64Index([1295], dtype='int64')
Foi
['terça-feira', 'quinta-feira']
begin:  11
begin1:  11
Int64Index([19], dtype='int64')
begin1:  12
Int64Index([20], dtype='int64')
begin1:  13
Int64Index([21], dtype='int64')
begin1:  11
Int64Index([49], dtype='int64')
begin1:  12
Int64Index([50], dtype='int64')
begin1:  13
Int64Index([51], dtype='int64')
Foi
['terça-feira', 'quinta-feira']
A sala  104 no horário 11 do dia  terça-feira tá ocupada
begin:  11
begin1:  11
Int64Index([94], dtype='int64')
begin1:  12
Int64Index([95], dtype='int64')
begin1:  11
Int64Index([124], dtype='int64')
begin1:  12
Int64Index([125], dtype='int64')
Foi
['terça-feira', 'quinta-feira']
A sala  1

['terça-feira', 'quinta-feira']
A sala  104 no horário 11 do dia  terça-feira tá ocupada
A sala  137 no horário 11 do dia  terça-feira tá ocupada
A sala  230A no horário 11 do dia  terça-feira tá ocupada
A sala  233 no horário 11 do dia  terça-feira tá ocupada
A sala  237B no horário 11 do dia  terça-feira tá ocupada
A sala  238 no horário 11 do dia  terça-feira tá ocupada
A sala  239 no horário 11 do dia  terça-feira tá ocupada
A sala  307A no horário 11 do dia  terça-feira tá ocupada
A sala  331 no horário 11 do dia  terça-feira tá ocupada
A sala  332 no horário 11 do dia  terça-feira tá ocupada
A sala  334-D no horário 11 do dia  terça-feira tá ocupada
A sala  336 no horário 11 do dia  terça-feira tá ocupada
A sala  337 no horário 11 do dia  terça-feira tá ocupada
A sala  339 no horário 11 do dia  terça-feira tá ocupada
A sala  443 no horário 11 do dia  terça-feira tá ocupada
A sala  443A no horário 11 do dia  terça-feira tá ocupada
A sala  444 no horário 11 do dia  terça-feira tá o

A sala  307A no horário 14 do dia  terça-feira tá ocupada
A sala  331 no horário 14 do dia  terça-feira tá ocupada
A sala  332 no horário 14 do dia  terça-feira tá ocupada
A sala  334-D no horário 14 do dia  terça-feira tá ocupada
A sala  336 no horário 14 do dia  terça-feira tá ocupada
begin:  14
begin1:  14
Int64Index([922], dtype='int64')
begin1:  15
Int64Index([923], dtype='int64')
begin1:  14
Int64Index([967], dtype='int64')
begin1:  15
Int64Index([968], dtype='int64')
Foi
['terça-feira', 'quinta-feira', 'sexta-feira']
A sala  104 no horário 14 do dia  terça-feira tá ocupada
A sala  137 no horário 14 do dia  terça-feira tá ocupada
A sala  230A no horário 14 do dia  terça-feira tá ocupada
A sala  233 no horário 14 do dia  terça-feira tá ocupada
A sala  237B no horário 14 do dia  terça-feira tá ocupada
A sala  238 no horário 14 do dia  terça-feira tá ocupada
A sala  239 no horário 14 do dia  terça-feira tá ocupada
A sala  307A no horário 14 do dia  terça-feira tá ocupada
A sala  331

A sala  230A no horário 14 do dia  terça-feira tá ocupada
A sala  233 no horário 14 do dia  terça-feira tá ocupada
A sala  237B no horário 14 do dia  terça-feira tá ocupada
A sala  238 no horário 14 do dia  terça-feira tá ocupada
A sala  239 no horário 14 do dia  terça-feira tá ocupada
A sala  307A no horário 14 do dia  terça-feira tá ocupada
A sala  331 no horário 14 do dia  terça-feira tá ocupada
A sala  332 no horário 14 do dia  terça-feira tá ocupada
A sala  334-D no horário 14 do dia  terça-feira tá ocupada
A sala  336 no horário 14 do dia  terça-feira tá ocupada
A sala  337 no horário 14 do dia  terça-feira tá ocupada
A sala  339 no horário 14 do dia  terça-feira tá ocupada
A sala  443 no horário 14 do dia  terça-feira tá ocupada
A sala  443A no horário 14 do dia  terça-feira tá ocupada
A sala  444 no horário 14 do dia  terça-feira tá ocupada
A sala  445 no horário 14 do dia  terça-feira tá ocupada
A sala  446-D no horário 14 do dia  terça-feira tá ocupada
A sala  447 no horário 

A sala  502C no horário 14 do dia  terça-feira tá ocupada
A sala  531B no horário 14 do dia  terça-feira tá ocupada
A sala  234A no horário 14 do dia  terça-feira tá ocupada
A sala  234B no horário 14 do dia  terça-feira tá ocupada
A sala  334-E no horário 14 do dia  terça-feira tá ocupada
begin:  14
begin1:  14
Int64Index([2197], dtype='int64')
begin1:  15
Int64Index([2198], dtype='int64')
begin1:  14
Int64Index([2227], dtype='int64')
begin1:  15
Int64Index([2228], dtype='int64')
Foi
['terça-feira', 'quinta-feira']
A sala  104 no horário 14 do dia  terça-feira tá ocupada
A sala  137 no horário 14 do dia  terça-feira tá ocupada
A sala  230A no horário 14 do dia  terça-feira tá ocupada
A sala  233 no horário 14 do dia  terça-feira tá ocupada
A sala  237B no horário 14 do dia  terça-feira tá ocupada
A sala  238 no horário 14 do dia  terça-feira tá ocupada
A sala  239 no horário 14 do dia  terça-feira tá ocupada
A sala  307A no horário 14 do dia  terça-feira tá ocupada
A sala  331 no horá

A sala  334-D no horário 16 do dia  terça-feira tá ocupada
A sala  336 no horário 16 do dia  terça-feira tá ocupada
A sala  337 no horário 16 do dia  terça-feira tá ocupada
A sala  339 no horário 16 do dia  terça-feira tá ocupada
begin:  16
begin1:  16
Int64Index([1074], dtype='int64')
begin1:  17
Int64Index([1075], dtype='int64')
begin1:  16
Int64Index([1104], dtype='int64')
begin1:  17
Int64Index([1105], dtype='int64')
Foi
['terça-feira', 'quinta-feira']
A sala  104 no horário 16 do dia  terça-feira tá ocupada
A sala  137 no horário 16 do dia  terça-feira tá ocupada
A sala  230A no horário 16 do dia  terça-feira tá ocupada
A sala  233 no horário 16 do dia  terça-feira tá ocupada
A sala  237B no horário 16 do dia  terça-feira tá ocupada
A sala  238 no horário 16 do dia  terça-feira tá ocupada
A sala  239 no horário 16 do dia  terça-feira tá ocupada
A sala  307A no horário 16 do dia  terça-feira tá ocupada
A sala  331 no horário 16 do dia  terça-feira tá ocupada
A sala  332 no horário 

begin1:  16
Int64Index([1704], dtype='int64')
begin1:  17
Int64Index([1705], dtype='int64')
begin1:  18
Int64Index([1706], dtype='int64')
begin1:  19
Int64Index([1707], dtype='int64')
Foi
['terça-feira', 'quinta-feira']
A sala  104 no horário 16 do dia  terça-feira tá ocupada
A sala  137 no horário 16 do dia  terça-feira tá ocupada
A sala  230A no horário 16 do dia  terça-feira tá ocupada
A sala  233 no horário 16 do dia  terça-feira tá ocupada
A sala  237B no horário 16 do dia  terça-feira tá ocupada
A sala  238 no horário 16 do dia  terça-feira tá ocupada
A sala  239 no horário 16 do dia  terça-feira tá ocupada
A sala  307A no horário 16 do dia  terça-feira tá ocupada
A sala  331 no horário 16 do dia  terça-feira tá ocupada
A sala  332 no horário 16 do dia  terça-feira tá ocupada
A sala  334-D no horário 16 do dia  terça-feira tá ocupada
A sala  336 no horário 16 do dia  terça-feira tá ocupada
A sala  337 no horário 16 do dia  terça-feira tá ocupada
A sala  339 no horário 16 do dia  

A sala  137 no horário 18 do dia  terça-feira tá ocupada
A sala  230A no horário 18 do dia  terça-feira tá ocupada
A sala  233 no horário 18 do dia  terça-feira tá ocupada
begin:  18
begin1:  18
Int64Index([326], dtype='int64')
begin1:  19
Int64Index([327], dtype='int64')
Foi
['terça-feira', 'quinta-feira']
A sala  104 no horário 18 do dia  terça-feira tá ocupada
A sala  137 no horário 18 do dia  terça-feira tá ocupada
A sala  230A no horário 18 do dia  terça-feira tá ocupada
A sala  233 no horário 18 do dia  terça-feira tá ocupada
A sala  237B no horário 18 do dia  terça-feira tá ocupada
begin:  18
begin1:  18
Int64Index([401], dtype='int64')
begin1:  19
Int64Index([402], dtype='int64')
begin1:  18
Int64Index([431], dtype='int64')
begin1:  19
Int64Index([432], dtype='int64')
Foi
['terça-feira', 'quinta-feira']
A sala  104 no horário 18 do dia  terça-feira tá ocupada
A sala  137 no horário 18 do dia  terça-feira tá ocupada
A sala  230A no horário 18 do dia  terça-feira tá ocupada
A sal

A sala  337 no horário 18 do dia  terça-feira tá ocupada
A sala  339 no horário 18 do dia  terça-feira tá ocupada
A sala  443 no horário 18 do dia  terça-feira tá ocupada
A sala  443A no horário 18 do dia  terça-feira tá ocupada
A sala  444 no horário 18 do dia  terça-feira tá ocupada
A sala  445 no horário 18 do dia  terça-feira tá ocupada
A sala  446-D no horário 18 do dia  terça-feira tá ocupada
begin:  18
begin1:  18
Int64Index([1451], dtype='int64')
begin1:  19
Int64Index([1452], dtype='int64')
begin1:  20
Int64Index([1453], dtype='int64')
begin1:  21
Int64Index([1454], dtype='int64')
Foi
['terça-feira', 'quinta-feira']
A sala  104 no horário 18 do dia  terça-feira tá ocupada
A sala  137 no horário 18 do dia  terça-feira tá ocupada
A sala  230A no horário 18 do dia  terça-feira tá ocupada
A sala  233 no horário 18 do dia  terça-feira tá ocupada
A sala  237B no horário 18 do dia  terça-feira tá ocupada
A sala  238 no horário 18 do dia  terça-feira tá ocupada
A sala  239 no horário 

A sala  332 no horário 9 do dia  quarta-feira tá ocupada
begin:  9
begin1:  9
Int64Index([782], dtype='int64')
begin1:  10
Int64Index([783], dtype='int64')
Foi
['quarta-feira']
A sala  104 no horário 9 do dia  quarta-feira tá ocupada
A sala  137 no horário 9 do dia  quarta-feira tá ocupada
A sala  230A no horário 9 do dia  quarta-feira tá ocupada
A sala  233 no horário 9 do dia  quarta-feira tá ocupada
A sala  237B no horário 9 do dia  quarta-feira tá ocupada
A sala  238 no horário 9 do dia  quarta-feira tá ocupada
A sala  239 no horário 9 do dia  quarta-feira tá ocupada
A sala  307A no horário 9 do dia  quarta-feira tá ocupada
A sala  331 no horário 9 do dia  quarta-feira tá ocupada
A sala  332 no horário 9 do dia  quarta-feira tá ocupada
A sala  334-D no horário 9 do dia  quarta-feira tá ocupada
begin:  9
begin1:  9
Int64Index([857], dtype='int64')
begin1:  10
Int64Index([858], dtype='int64')
begin1:  11
Int64Index([859], dtype='int64')
Foi
['segunda-feira', 'quarta-feira']
A sala  1

A sala  104 no horário 13 do dia  quarta-feira tá ocupada
A sala  137 no horário 13 do dia  quarta-feira tá ocupada
A sala  230A no horário 13 do dia  quarta-feira tá ocupada
A sala  233 no horário 13 do dia  quarta-feira tá ocupada
A sala  237B no horário 13 do dia  quarta-feira tá ocupada
begin:  13
begin1:  13
Int64Index([411], dtype='int64')
begin1:  14
Int64Index([412], dtype='int64')
begin1:  15
Int64Index([413], dtype='int64')
Foi
['quarta-feira']
A sala  104 no horário 13 do dia  quarta-feira tá ocupada
A sala  137 no horário 13 do dia  quarta-feira tá ocupada
A sala  230A no horário 13 do dia  quarta-feira tá ocupada
A sala  233 no horário 13 do dia  quarta-feira tá ocupada
A sala  237B no horário 13 do dia  quarta-feira tá ocupada
A sala  238 no horário 13 do dia  quarta-feira tá ocupada
begin:  13
begin1:  13
Int64Index([486], dtype='int64')
Foi
['quarta-feira']
A sala  104 no horário 14 do dia  quarta-feira tá ocupada
A sala  137 no horário 14 do dia  quarta-feira tá ocupad

A sala  445 no horário 14 do dia  segunda-feira tá ocupada
A sala  446-D no horário 14 do dia  segunda-feira tá ocupada
A sala  447 no horário 14 do dia  segunda-feira tá ocupada
A sala  448 no horário 14 do dia  segunda-feira tá ocupada
A sala  449 no horário 14 do dia  segunda-feira tá ocupada
A sala  450 no horário 14 do dia  segunda-feira tá ocupada
A sala  502 no horário 14 do dia  segunda-feira tá ocupada
A sala  502C no horário 14 do dia  segunda-feira tá ocupada
A sala  531B no horário 14 do dia  segunda-feira tá ocupada
A sala  234A no horário 14 do dia  segunda-feira tá ocupada
A sala  234B no horário 14 do dia  segunda-feira tá ocupada
A sala  334-E no horário 14 do dia  segunda-feira tá ocupada
A sala  341 no horário 14 do dia  segunda-feira tá ocupada
A sala  428 no horário 14 do dia  segunda-feira tá ocupada
A sala  446-E no horário 14 do dia  segunda-feira tá ocupada
begin:  14
begin1:  14
Int64Index([2407], dtype='int64')
begin1:  15
Int64Index([2408], dtype='int64')
be

A sala  137 no horário 16 do dia  segunda-feira tá ocupada
A sala  230A no horário 16 do dia  segunda-feira tá ocupada
A sala  233 no horário 16 do dia  segunda-feira tá ocupada
A sala  237B no horário 16 do dia  segunda-feira tá ocupada
A sala  238 no horário 16 do dia  segunda-feira tá ocupada
A sala  239 no horário 16 do dia  segunda-feira tá ocupada
A sala  307A no horário 16 do dia  segunda-feira tá ocupada
A sala  331 no horário 16 do dia  segunda-feira tá ocupada
A sala  332 no horário 16 do dia  segunda-feira tá ocupada
A sala  334-D no horário 16 do dia  segunda-feira tá ocupada
A sala  336 no horário 16 do dia  segunda-feira tá ocupada
A sala  337 no horário 16 do dia  segunda-feira tá ocupada
A sala  339 no horário 16 do dia  segunda-feira tá ocupada
A sala  443 no horário 16 do dia  segunda-feira tá ocupada
A sala  443A no horário 16 do dia  segunda-feira tá ocupada
A sala  444 no horário 16 do dia  segunda-feira tá ocupada
A sala  445 no horário 16 do dia  segunda-feira tá

A sala  104 no horário 7 do dia  quinta-feira tá ocupada
A sala  137 no horário 7 do dia  quinta-feira tá ocupada
A sala  230A no horário 7 do dia  quinta-feira tá ocupada
A sala  233 no horário 7 do dia  quinta-feira tá ocupada
begin:  7
begin1:  7
Int64Index([345], dtype='int64')
begin1:  8
Int64Index([346], dtype='int64')
Foi
['quinta-feira']
A sala  104 no horário 7 do dia  quinta-feira tá ocupada
A sala  137 no horário 7 do dia  quinta-feira tá ocupada
A sala  230A no horário 7 do dia  quinta-feira tá ocupada
A sala  233 no horário 7 do dia  quinta-feira tá ocupada
A sala  237B no horário 7 do dia  quinta-feira tá ocupada
begin:  7
begin1:  7
Int64Index([420], dtype='int64')
begin1:  8
Int64Index([421], dtype='int64')
begin1:  9
Int64Index([422], dtype='int64')
begin1:  10
Int64Index([423], dtype='int64')
Foi
['quinta-feira']
A sala  104 no horário 7 do dia  quinta-feira tá ocupada
A sala  137 no horário 7 do dia  quinta-feira tá ocupada
A sala  230A no horário 7 do dia  quinta-fe

A sala  331 no horário 11 do dia  quinta-feira tá ocupada
A sala  332 no horário 11 do dia  quinta-feira tá ocupada
A sala  334-D no horário 11 do dia  quinta-feira tá ocupada
A sala  336 no horário 11 do dia  quinta-feira tá ocupada
A sala  337 no horário 11 do dia  quinta-feira tá ocupada
A sala  339 no horário 11 do dia  quinta-feira tá ocupada
A sala  443 no horário 11 do dia  quinta-feira tá ocupada
A sala  443A no horário 11 do dia  quinta-feira tá ocupada
A sala  444 no horário 11 do dia  quinta-feira tá ocupada
A sala  445 no horário 11 do dia  quinta-feira tá ocupada
A sala  446-D no horário 11 do dia  quinta-feira tá ocupada
A sala  447 no horário 11 do dia  quinta-feira tá ocupada
A sala  448 no horário 11 do dia  quinta-feira tá ocupada
A sala  449 no horário 11 do dia  quinta-feira tá ocupada
begin:  11
begin1:  11
Int64Index([1699], dtype='int64')
begin1:  12
Int64Index([1700], dtype='int64')
begin1:  13
Int64Index([1701], dtype='int64')
Foi
['quinta-feira']
A sala  104 n

A sala  237B no horário 16 do dia  terça-feira tá ocupada
A sala  238 no horário 16 do dia  terça-feira tá ocupada
A sala  239 no horário 16 do dia  terça-feira tá ocupada
A sala  307A no horário 16 do dia  terça-feira tá ocupada
A sala  331 no horário 16 do dia  terça-feira tá ocupada
A sala  332 no horário 16 do dia  terça-feira tá ocupada
A sala  334-D no horário 16 do dia  terça-feira tá ocupada
A sala  336 no horário 16 do dia  terça-feira tá ocupada
A sala  337 no horário 16 do dia  terça-feira tá ocupada
A sala  339 no horário 16 do dia  terça-feira tá ocupada
A sala  443 no horário 16 do dia  terça-feira tá ocupada
A sala  443A no horário 16 do dia  terça-feira tá ocupada
A sala  444 no horário 16 do dia  terça-feira tá ocupada
A sala  445 no horário 16 do dia  terça-feira tá ocupada
A sala  446-D no horário 16 do dia  terça-feira tá ocupada
A sala  447 no horário 16 do dia  terça-feira tá ocupada
A sala  448 no horário 16 do dia  terça-feira tá ocupada
A sala  449 no horário 1

A sala  448 no horário 18 do dia  segunda-feira tá ocupada
begin:  18
begin1:  18
Int64Index([1586], dtype='int64')
begin1:  19
Int64Index([1587], dtype='int64')
begin1:  20
Int64Index([1588], dtype='int64')
begin1:  21
Int64Index([1589], dtype='int64')
begin1:  18
Int64Index([1631], dtype='int64')
begin1:  19
Int64Index([1632], dtype='int64')
begin1:  20
Int64Index([1633], dtype='int64')
begin1:  21
Int64Index([1634], dtype='int64')
Foi
['quinta-feira']
begin:  20
begin1:  20
Int64Index([58], dtype='int64')
begin1:  21
Int64Index([59], dtype='int64')
Foi
['sexta-feira']
A sala  104 no horário 7 do dia  sexta-feira tá ocupada
begin:  7
begin1:  7
Int64Index([135], dtype='int64')
begin1:  8
Int64Index([136], dtype='int64')
Foi
['sexta-feira']
A sala  104 no horário 7 do dia  sexta-feira tá ocupada
A sala  137 no horário 7 do dia  sexta-feira tá ocupada
begin:  7
begin1:  7
Int64Index([210], dtype='int64')
begin1:  8
Int64Index([211], dtype='int64')
begin1:  9
Int64Index([212], dtype='in

begin1:  13
Int64Index([66], dtype='int64')
Foi
['sexta-feira']
A sala  104 no horário 13 do dia  sexta-feira tá ocupada
begin:  13
begin1:  13
Int64Index([141], dtype='int64')
begin1:  14
Int64Index([142], dtype='int64')
begin1:  15
Int64Index([143], dtype='int64')
Foi
['sexta-feira']
begin:  14
begin1:  14
Int64Index([67], dtype='int64')
begin1:  15
Int64Index([68], dtype='int64')
begin1:  16
Int64Index([69], dtype='int64')
begin1:  17
Int64Index([70], dtype='int64')
Foi
['sexta-feira']
A sala  104 no horário 14 do dia  sexta-feira tá ocupada
A sala  137 no horário 14 do dia  sexta-feira tá ocupada
begin:  14
begin1:  14
Int64Index([217], dtype='int64')
begin1:  15
Int64Index([218], dtype='int64')
begin1:  16
Int64Index([219], dtype='int64')
begin1:  17
Int64Index([220], dtype='int64')
Foi
['sexta-feira']
A sala  104 no horário 14 do dia  sexta-feira tá ocupada
A sala  137 no horário 14 do dia  sexta-feira tá ocupada
A sala  230A no horário 14 do dia  sexta-feira tá ocupada
begin:  1

In [56]:
df_tempo.head(20)

0   Dia da semana  Horário Sala Bloco Dept Capac. Departamento Disciplina  \
0   segunda-feira        7  104     D  NaN     60          TEE   TEE00114   
1   segunda-feira        8  104     D  NaN     60          TEE   TEE00114   
2   segunda-feira        9  104     D  NaN     60          TEC   TEC00224   
3   segunda-feira       10  104     D  NaN     60          TEC   TEC00224   
4   segunda-feira       11  104     D  NaN     60          TEE   TEE00115   
5   segunda-feira       12  104     D  NaN     60          TEE   TEE00115   
6   segunda-feira       13  104     D  NaN     60          TEC   TEC00216   
7   segunda-feira       14  104     D  NaN     60          TEC   TEC00216   
8   segunda-feira       15  104     D  NaN     60          TEC   TEC00216   
9   segunda-feira       16  104     D  NaN     60          TEC   TEC00216   
10  segunda-feira       17  104     D  NaN     60          TER   TER00039   
11  segunda-feira       18  104     D  NaN     60          TER   TER00039   
12  segunda-feira       19  104     D  NaN     60          TER   TER00039   
13  segunda-feira       20  104     D  NaN     60          TEC   TEC00273   
14  segunda-feira       21  104     D  NaN     60          TEC   TEC00273   
15    terça-feira        7  104     D  NaN     60          TEC   TEC00045   
16    terça-feira        8  104     D  NaN     60          TEC   TEC00045   
17    terça-feira        9  104     D  NaN     60          TEC   TEC00204   
18    terça-feira       10  104     D  NaN     60          TEC   TEC00204   
19    terça-feira       11  104     D  NaN     60          TEC   TEC00205   

0                         Professor  
0                         Alexandre  
1                         Alexandre  
2        MARCO AURELIO CHAVES FERRO  
3        MARCO AURELIO CHAVES FERRO  
4   Marcio Guimaraens e André Pinho  
5   Marcio Guimaraens e André Pinho  
6                    JULIANA PESSIN  
7                    JULIANA PESSIN  
8                    JULIANA PESSIN  
9                    JULIANA PESSIN  
10               Gabriel Nascimento  
11               Gabriel Nascimento  
12               Gabriel Nascimento  
13             RUBENEI NOVAIS SOUZA  
14             RUBENEI NOVAIS SOUZA  
15   ANDREIA ABREU DINIZ DE ALMEIDA  
16   ANDREIA ABREU DINIZ DE ALMEIDA  
17         CLAUDIO RIBEIRO CARVALHO  
18         CLAUDIO RIBEIRO CARVALHO  
19    ELIANE FERNANDES CORTES PIRES

In [49]:
df_tempo[(df_tempo['Professor'] == 'Alexandre')&(df_tempo['Sala'] == 104)]

0   Dia da semana  Horário Sala Bloco Dept Capac. Departamento Disciplina  \
0   segunda-feira        7  104     D  NaN     60          TEE   TEE00114   
1   segunda-feira        8  104     D  NaN     60          TEE   TEE00114   
30   quarta-feira        7  104     D  NaN     60          TEE   TEE00114   
31   quarta-feira        8  104     D  NaN     60          TEE   TEE00114   
60    sexta-feira        7  104     D  NaN     60          TEE   TEE00114   
61    sexta-feira        8  104     D  NaN     60          TEE   TEE00114   

0   Professor  
0   Alexandre  
1   Alexandre  
30  Alexandre  
31  Alexandre  
60  Alexandre  
61  Alexandre

In [65]:
df_aulas.iloc[0, 12] = 144

In [ ]:
df_aulas.iloc[0, 12]

In [77]:
df_aulas['Local'].value_counts()

104      25
137      22
233      20
230A     19
238      18
237B     15
239      14
334-D    14
307A     13
336      13
331      12
332      12
337      12
446-D    11
444      10
339      10
445       9
443A      9
443       9
447       8
448       8
449       6
450       5
502       5
502C      5
531B      5
334-E     3
234A      3
234B      3
341       2
428       2
446-E     2
201       2
202       1
204       1
Name: Local, dtype: int64